### DISCUSSION TITLE
Instead of downloading pertained model from hosted, how can I download the pertained models to local

### DISCUSSION THREAD

**Discussion by madhavi1102** [mindee/doctr#1350] [Q&A] [answered]
while instantiating ocr_predictor() model with pretrained models 'db_resnet50' and 'crnn_vgg16_bn' which get downloaded from public hosted. 
Instead of it, how can I download these pretrained models locally and refer to local path for these models while instantiating ocr_predictor.

Thanks in advance,
Madhavi

**Comment by felixdittrich92** [comment, marked as answer]
Hi @madhavi1102 👋,

Take a look at https://mindee.github.io/doctr/using_doctr/custom_models_training.html#loading-your-custom-trained-model how to load from local :)

The already downloaded models are stored in `.cache/doctr/models` you can copy paste it from there

**Reply by madhavi1102** [reply]
Hi @felixdittrich92 ,

Got it, Thanks for the quick response :)

**Reply by yumikim381** [reply]
@felixdittrich92 Hi, i couldn't find the folder, is the location still accurate?

**Comment by aryakp2302** [comment]
> while instantiating ocr_predictor() model with pretrained models 'db_resnet50' and 'crnn_vgg16_bn' which get downloaded from public hosted. Instead of it, how can I download these pretrained models locally and refer to local path for these models while instantiating ocr_predictor.
> 
> Thanks in advance, Madhavi

Have you find any solution because i also wanted to use pretrained model locally...it would be really great if you can help me                                            Thank you In advance, Arya Patel

**Reply by felixT2K** [reply]
Hi @madhavi1102 👋🏼,

If you have already downloaded the checkpoints they are stored in `/home/.cache/doctr/models` otherwise you can also download specific models from the url

for example db_resnet50 pytorch:  
https://github.com/mindee/doctr/blob/0d849152a852cb55b3fb0cc0e5e602600349d97d/doctr/models/detection/differentiable_binarization/pytorch.py#L30

**Reply by yumikim381** [reply]
Thank you so much!!

**Comment by madhavi1102** [comment]
Hi T2K-Felix,


Thats great, thanks for letting me know.

Regards,
Madhavi

**Comment by Sonsal** [comment]
@felixdittrich92 Hello! Seems like docs a little bit outdated and .load_weights method is deprecated now.
Working solution for me (example for detection model, should work for recognition model too):
```python
from doctr.models import ocr_predictor, db_resnet50

PATH_TO_DET_MODEL = <path_to_.pt_weights_file>

det_model = db_resnet50(pretrained=False, pretrained_backbone=False)
det_model.load_state_dict(torch.load(PATH_TO_DET_MODEL, weights_only=True))
```

**Reply by felixdittrich92** [reply]
Hi @Sonsal 😊
You mean because of `weights_only=True` ? That's only a warning from PyTorch because the default value will change in a future release - you can ignore it - it shouldn't matter if the value is True or False :)

**Reply by felixdittrich92** [reply]
`load_weights` is by the way for Tensorflow :D you have two tabs in the docs one for Tensorflow and the other for PyTorch ^^
